### PR TITLE
Add Leptos Alert

### LINKED ISSUE #62: Add Alert for Leptos
- [ ] Component
- [ ] Docs

### PR DESCRIPTION
**Pull request by israelsilvabarbara**
Closes #62.

Added Alert feature to Leptos and also added examples in book-example.

### CODE REVIEW

**Comment by DanielleHuisman** [comment]
Could you please resolve the merge conflicts? I will review the PR this weekend.

File: book-examples/leptos/src/default/alert.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+};
+
+#[component(transparent)]
+pub fn AlertRoutes() -> impl MatchNestedRoutes + Clone {
+    view! {
+        <ParentRoute path=path!("/alert") view=Outlet>
+            <Route path=path!("/") view=alert::AlertDemo />
+            <Route path=path!("/destructive") view=alert_destructive::AlertDestructive />
+            <Route path=path!("/icon") view=alert_icon::AlertIcon />
+            <Route path=path!("/class") view=alert_class::AlertClass />
```
Comment (DanielleHuisman) [inline book-examples/leptos/src/default/alert.rs:20]: These examples don't exist for shadcn/ui's alert. The book only ports existing shadcn/ui examples.

File: book-examples/leptos/src/default.rs [hunk 2]
```diff
@@ -2,6 +2,10 @@ mod components;
 
 #[cfg(feature = "button")]
 mod button;
+
+#[cfg(feature = "alert")]
+mod alert;
+
 #[cfg(feature = "card")]
 mod card;
```
Comment (DanielleHuisman) [inline book-examples/leptos/src/default.rs:10]:
```suggestion
#[cfg(feature = "alert")]
mod alert;
#[cfg(feature = "button")]
mod button;
#[cfg(feature = "card")]
mod card;
```

File: book-examples/leptos/src/default.rs [hunk 3]
```diff
@@ -18,6 +22,10 @@ pub fn Default() -> impl MatchNestedRoutes + Clone {
         {
             component_view(self::button::ButtonRoutes, ())
         },
+        #[cfg(feature = "alert")]
```
Comment (DanielleHuisman) [inline book-examples/leptos/src/default.rs:25]: Please keep this list sorted alphabetically.

File: book-examples/leptos/src/new_york.rs [hunk 4]
```diff
@@ -2,6 +2,10 @@ mod components;
 
 #[cfg(feature = "button")]
 mod button;
+
+#[cfg(feature = "alert")]
+mod alert;
+
 #[cfg(feature = "card")]
 mod card;
```
Comment (DanielleHuisman) [inline book-examples/leptos/src/new_york.rs:10]:
```suggestion
#[cfg(feature = "alert")]
mod alert;
#[cfg(feature = "button")]
mod button;
#[cfg(feature = "card")]
mod card;
```

File: book-examples/leptos/src/new_york.rs [hunk 5]
```diff
@@ -18,6 +22,10 @@ pub fn NewYork() -> impl MatchNestedRoutes + Clone {
         {
             component_view(self::button::ButtonRoutes, ())
         },
+        #[cfg(feature = "alert")]
```
Comment (DanielleHuisman) [inline book-examples/leptos/src/new_york.rs:25]: Please keep this list sorted alphabetically.

File: book-examples/leptos/src/default/components/ui.rs [hunk 6]
```diff
@@ -2,5 +2,9 @@
 
 #[cfg(any(feature = "button", feature = "card"))]
 pub use shadcn_ui_leptos_button::default as button;
+
+#[cfg(feature = "alert")]
+pub use shadcn_ui_leptos_alert::default as alert;
+
 #[cfg(feature = "card")]
 pub use shadcn_ui_leptos_card::default as card;
```
Comment (DanielleHuisman) [inline book-examples/leptos/src/default/components/ui.rs:10]:
```suggestion
#[cfg(feature = "alert")]
pub use shadcn_ui_leptos_alert::default as alert;
#[cfg(any(feature = "button", feature = "card"))]
pub use shadcn_ui_leptos_button::default as button;
#[cfg(feature = "card")]
pub use shadcn_ui_leptos_card::default as card;
```

File: book-examples/leptos/src/new_york/components/ui.rs [hunk 7]
```diff
@@ -2,5 +2,9 @@
 
 #[cfg(any(feature = "button", feature = "card"))]
 pub use shadcn_ui_leptos_button::new_york as button;
+
+#[cfg(feature = "alert")]
+pub use shadcn_ui_leptos_alert::new_york as alert;
+
 #[cfg(feature = "card")]
 pub use shadcn_ui_leptos_card::new_york as card;
```
Comment (DanielleHuisman) [inline book-examples/leptos/src/new_york/components/ui.rs:10]:
```suggestion
#[cfg(feature = "alert")]
pub use shadcn_ui_leptos_alert::new_york as alert;
#[cfg(any(feature = "button", feature = "card"))]
pub use shadcn_ui_leptos_button::new_york as button;
#[cfg(feature = "card")]
pub use shadcn_ui_leptos_card::new_york as card;
```

File: book-examples/leptos/Cargo.toml [hunk 8]
```diff
@@ -33,8 +36,11 @@ button = [
     "radix-leptos-icons/envelope-open",
     "radix-leptos-icons/reload",
 ]
+alert = [
+    "dep:shadcn-ui-leptos-alert"
+]
```
Comment (DanielleHuisman) [inline book-examples/leptos/Cargo.toml:41]: Please keep this sorted alphabetically.

When using icons in examples, the optional dependency needs to be enabled here, see the Yew book examples.

**Review by DanielleHuisman — CHANGES_REQUESTED** [review]
Thank you for contributing! The component is coming along nicely.

**Comment by israelsilvabarbara** [comment]
To make it easier to review, I have implemented the requested changes in separate commits.

**Review by DanielleHuisman — APPROVED** [review]

**Pull request merged**